plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveS3AndGlueMetastoreTest.java

```diff
@@ -254,14 +253,55 @@ public void testCtasWithIncorrectLocation()
                 .hasStackTraceContaining("Fragment is not allowed in a file system location");
     }
 
+    @Test
+    public void testCreateTableWithDoubleSlash()
+    {
+        String schemaName = "test_create_table_with_double_slash_" + randomNameSuffix();
+        String schemaLocation = "s3://%s/%s/double_slash//test_schema".formatted(bucketName, schemaName);
+        String tableName = "test_create_table_with_double_slash_" + randomNameSuffix();
+        String schemaTableName = schemaName + "." + tableName;
+
+        // Previously, HiveLocationService replaced double slash with single slash
+        assertUpdate("CREATE SCHEMA " + schemaName + " WITH (location = '" + schemaLocation + "')");
+        String existingKey = "%s/double_slash/test_schema/%s".formatted(schemaName, tableName);
+        s3.putObject(bucketName, existingKey, "test content");
+
+        assertUpdate("CREATE TABLE " + schemaTableName + "(col_int int)");
+        assertUpdate("INSERT INTO " + schemaTableName + " VALUES 1", 1);
+        assertQuery("SELECT * FROM " + schemaTableName, "VALUES 1");
+        assertUpdate("DROP TABLE " + schemaTableName);
+        s3.deleteObject(bucketName, existingKey);
+    }
+
+    @Test
+    public void testCtasWithDoubleSlash()
+    {
+        String schemaName = "test_ctas_with_double_slash_" + randomNameSuffix();
+        String schemaLocation = "s3://%s/%s/double_slash//test_schema".formatted(bucketName, schemaName);
+        String tableName = "test_create_table_with_double_slash_" + randomNameSuffix();
+        String schemaTableName = schemaName + "." + tableName;
+
+        // Previously, HiveLocationService replaced double slash with single slash
+        assertUpdate("CREATE SCHEMA " + schemaName + " WITH (location = '" + schemaLocation + "')");
+        String existingKey = "%s/double_slash/test_schema/%s".formatted(schemaName, tableName);
+        s3.putObject(bucketName, existingKey, "test content");
+
+        assertUpdate("CREATE TABLE " + schemaTableName + " AS SELECT 1 AS col_int", 1);
+        assertQuery("SELECT * FROM " + schemaTableName, "VALUES 1");
+        assertUpdate("DROP TABLE " + schemaTableName);
+        s3.deleteObject(bucketName, existingKey);
+    }
```

Comment:
Are these new tests needed?
When working on https://github.com/trinodb/trino/pull/17958 i  was under impression that other existing tests (those with data provider) cover these cases.

Reply:
These tests are needed for changes in `HiveLocationService`. Existing tests don't cover this scenario.

Reply:
`testBasicOperationsWithProvidedSchemaLocation` exercises `HiveLocationService`
is it missing more coverage?

Reply:
`testBasicOperationsWithProvidedSchemaLocation` doesn't cover the test when the normalized table location already exists in S3. This test puts an object on single-slash table location and then creates a table on double-slash location. `HiveLocationService` threw an incorrect already-exists error because the class internally replaced double-slash with single-slash. 

It wasn't a silent correctness issue, so no need to fix it in this release, but it would be nice to add the test and improve it later.

Reply:
agreed.
let's have explanatory comment in the test. it seems at least for me understanding its importance wasn't easy